addons/project/models/project.py

```diff
@@ -1805,12 +1805,13 @@ def default_get(self, default_fields):
                 )
                 if partner_id:
                     vals['partner_id'] = partner_id
-        if vals.get('project_id'):
-            project = self.env['project.project'].browse(vals.get('project_id'))
+        project_id = vals.get('project_id', self.env.context.get('default_project_id'))
+        if project_id:
+            project = self.env['project.project'].browse(project_id)
             if project.analytic_account_id:
                 vals['analytic_account_id'] = project.analytic_account_id.id
             if project.analytic_tag_ids:
-                vals['analytic_tag_ids'] = [Command.set(project.analytic_tag_ids.ids)]
+                vals['analytic_tag_ids'] = [Command.link(tag.id) for tag in project.analytic_tag_ids]
```

Comment:
If I am not mistaken this is already made [here](https://github.com/odoo/odoo/pull/94840).
And the following change is not necessary in my opinion.
```diff
-                vals['analytic_tag_ids'] = [Command.set(project.analytic_tag_ids.ids)]
+                vals['analytic_tag_ids'] = [Command.link(tag.id) for tag in project.analytic_tag_ids]
```